src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -178,4 +211,526 @@ export class NetworkAreaDiagramViewer {
+        const isTransformerEdge =
+            edgeType == DiagramUtils.EdgeType.TWO_WINDINGS_TRANSFORMER ||
+            edgeType == DiagramUtils.EdgeType.PHASE_SHIFT_TRANSFORMER;
+        const isHVDCLineEdge = edgeType == DiagramUtils.EdgeType.HVDC_LINE;
+        this.moveHalfEdge(
+            edgeNode,
+            '1',
+            edgeStart1,
+            edgeFork1,
+            edgeMiddle,
+            isTransformerEdge
+        );
+        this.moveHalfEdge(
+            edgeNode,
+            '2',
+            edgeStart2,
+            edgeFork2,
+            edgeMiddle,
+            isTransformerEdge
+        );
+        if (isTransformerEdge) {
+            this.moveTransformer(
+                edgeNode,
+                edgeFork1 == null ? edgeStart1 : edgeFork1,
+                edgeMiddle,
+                edgeFork2 == null ? edgeStart2 : edgeFork2,
+                edgeMiddle,
+                edgeType
+            );
+        } else if (isHVDCLineEdge) {
+            this.moveConverterStation(
+                edgeNode,
+                edgeFork1 == null ? edgeStart1 : edgeFork1,
+                edgeMiddle,
+                edgeFork2 == null ? edgeStart2 : edgeFork2,
+                edgeMiddle
+            );
+        }
+    }
+
+    private moveHalfEdge(
+        edgeNode: SVGGraphicsElement,
+        side: string,
+        startPolyline: Point,
+        middlePolyline: Point | null, // if null -> straight line
+        endPolyline: Point,
+        transformerEdge: boolean
+    ) {
+        // get half edge element
+        const halfEdge: SVGGraphicsElement | null = edgeNode.querySelector(
+            "[id='" + edgeNode.id + '.' + side + "']"
+        );
+        // move edge polyline
+        const polyline: SVGGraphicsElement | null | undefined =
+            halfEdge?.querySelector('polyline');
+        // if transformer edge reduce edge polyline, leaving space for the transformer
+        endPolyline = transformerEdge
+            ? DiagramUtils.getDistance(
+                  endPolyline,
+                  middlePolyline == null ? startPolyline : middlePolyline,
+                  1.5 * this.svgParameters.getTransfomerCircleRadius()
+              )
+            : endPolyline;
+        const polylinePoints: string =
+            middlePolyline == null
+                ? startPolyline.x.toFixed(2) +
+                  ',' +
+                  startPolyline.y.toFixed(2) +
+                  ' ' +
+                  endPolyline.x.toFixed(2) +
+                  ',' +
+                  endPolyline.y.toFixed(2)
+                : startPolyline.x.toFixed(2) +
+                  ',' +
+                  startPolyline.y.toFixed(2) +
+                  ' ' +
+                  middlePolyline.x.toFixed(2) +
+                  ',' +
+                  middlePolyline.y.toFixed(2) +
+                  ' ' +
+                  endPolyline.x.toFixed(2) +
+                  ',' +
+                  endPolyline.y;
```

Comment:
`.toFixed(2)` is missing

Could you factorize the common start / end here? well, only if it makes the code more clear!

Reply:
string creation rewritten